core/src/gl/mesh.cpp

```diff
@@ -37,17 +37,21 @@ MeshBase::~MeshBase() {
     // Deleting a index/array buffer being used ends up setting up the current vertex/index buffer to 0
     // after the driver finishes using it, force the render state to be 0 for vertex/index buffer
 
-    if (m_glVertexBuffer) {
-        if (RenderState::vertexBuffer.compare(m_glVertexBuffer)) {
-            RenderState::vertexBuffer.init(0, false);
+    if (RenderState::isValidGeneration(m_generation)) {
```

Comment:
Yes, only textures were a special case for the caches in RasterSource so far. At some point we could have a GLResource class that holds the gl handle and moves things for destruction to the mainloop unless it's called on the mainloop already.